spring-cloud-stream-modules-docs/src/main/asciidoc/sinks.adoc

```diff
@@ -537,6 +537,69 @@ $$key$$:: $$name for the key$$ *($$String$$, no default)*
 $$queue$$:: $$name for the queue$$ *($$String$$, no default)*
 $$topic$$:: $$name for the topic$$ *($$String$$, no default)*
 
+[[spring-cloud-stream-modules-sink-router]]
+=== Dynamic Router (`router`)
+
+The Dynamic Router support allows for routing messages to *named destinations* based on the evaluation of a SpEL
+expression or Groovy Script.
+
+==== SpEL-based Routing
+
+The expression evaluates against the message and returns either a channel name, or the key to a map of channel names.
+
+For more information, please see the "Routers and the Spring Expression Language (SpEL)" subsection in the Spring
+Integration Reference manual
+http://docs.spring.io/spring-integration/reference/html/messaging-routing-chapter.html#router-namespace[Configuring (Generic) Router section].
+
+==== Groovy-based Routing
+
+Instead of SpEL expressions, Groovy scripts can also be used. Let's create a Groovy script in the file system at
+"file:/my/path/router.groovy", or "classpath:/my/path/router.groovy" :
+
+[source,groovy]
+----
+println("Groovy processing payload '" + payload +"'");
+if (payload.contains('a')) {
+    return "foo"
+}
+else {
+    return "bar"
+}
+----
+
+If you want to pass variable values to your script, you can statically bind values using the _variables_ option or
+optionally pass the path to a properties file containing the bindings using the _propertiesLocation_ option.
+All properties in the file will be made available to the script as variables. You may specify both _variables_ and
+_propertiesLocation_, in which case any duplicate values provided as _variables_ override values provided in
+_propertiesLocation_.
+Note that _payload_ and _headers_ are implicitly bound to give you access to the data contained in a message.
+
+For more information, see the Spring Integration Reference manual
+http://docs.spring.io/spring-integration/reference/html/messaging-endpoints-chapter.html#groovy[Groovy Support].
+
+==== Options
+
+The **$$router$$** $$sink$$ has the following options:
+
+$$destinations$$:: $$comma-delimited destinations mapped from evaluation results$$ *($$String$$, no default)*
+$$defaultOutputChannel$$:: $$Where to route messages where the channel cannot be resolved$$ *($$String$$, default: `nullChannel`)*
+$$expression$$:: $$a SpEL expression used to determine the destination$$ *($$String$$, default: `headers['routeTo']`)*
+$$propertiesLocation$$:: $$the path of a properties file containing custom script variable bindings$$ *($$String$$, no default)*
+$$refreshDelay$$:: $$how often to check (in milliseconds) whether the script (if present) has changed; -1 for never$$ *($$long$$, default: `60000`)*
+$$script$$:: $$reference to a script used to process messages$$ *($$String$$, no default)*
+$$destinationMapping$$:: $$Destination mappings as a new line delimited string of name-value pairs, e.g. 'foo=bar\n baz=car'.$$ *($$String$$, no default)*
```

Comment:
Well, looks like we agreed for the "destinationMapping**s**" name.
Will fix on merge.